### PR TITLE
BUG: Fix bug with scale bar overplotting

### LINKED ISSUE #10128: Scaling label overplotting in Raw.plot()
MWE:
```python
# %%
from pathlib import Path
import mne


sample_dir = Path(mne.datasets.sample.data_path())
sample_fname = sample_dir / 'MEG' / 'sample' / 'sample_audvis_raw.fif'

raw = mne.io.read_raw_fif(sample_fname)
raw.plot(butterfly=True, start=0, duration=1)
```
Take a look at the scaling label of the `grad` channels:

Interestingly, when only plotting MEG channels, this problem does not appear. Also the gray "background noise" is gone:
```
# %%
raw.pick_types(meg=True)
raw.plot(butterfly=True, start=0, duration=1)
```

cc @drammock 

```
Platform:       macOS-12.1-x86_64-i386-64bit
Python:         3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 20:33:18)  [Clang 11.1.0 ]
Executable:     /Users/hoechenberger/miniforge3/envs/mne/bin/python
CPU:            i386: 8 cores
Memory:         16.0 GB

mne:            1.0.dev0
numpy:          1.21.4 {blas=NO_ATLAS_INFO, lapack=lapack}
scipy:          1.7.3
matplotlib:     3.5.0 {backend=module://matplotlib_inline.backend_inline}

sklearn:        1.0.1
numba:          0.53.1
nibabel:        3.2.1
nilearn:        0.8.1
dipy:           1.4.1
cupy:           Not found
pandas:         1.3.4
pyvista:        0.32.1 {OpenGL 4.1 INTEL-18.3.5 via Intel(R) Iris(TM) Plus Graphics OpenGL Engine}
pyvistaqt:      0.5.0
ipyvtklink:     0.2.1
vtk:            9.0.3
PyQt5:          5.12.3
ipympl:         0.8.2
mne_qt_browser: 0.1.7
pooch:          v1.5.2
```

### PR DESCRIPTION
**Pull request by larsoner**
Closes #10128

Test fails on `main` and passes on this PR. And locally testing visually I now get:
```
MNE_BROWSER_BACKEND=matplotlib mne browse_raw ~/mne_data/MNE-sample-data/MEG/sample/sample_audvis_raw.fif --butterfly
```

Screenshot from 2024-07-16 18-10-40

Also adds a `--butterfly` option to `mne browse_raw` but I think this is probably not worth adding to the changelog because it's so minor.

### CODE REVIEW

File: mne/viz/tests/test_raw.py [hunk 1]
```diff
@@ -1214,3 +1219,35 @@ def test_plotting_memory_garbage_collection(raw, pg_backend):
...
+        qtbot.wait_exposed(fig)
+        for _ in range(10):
+            ch_types = list(fig.mne.channel_axis.ch_texts)  # keys
+            if len(ch_types) > 0:
+                break
+            qtbot.wait(100)  # pragma: no cover
+        # the grad/mag difference here is intentional in _pg_figure.py
+        assert ch_types == ["grad", "mag", "eeg", "eog", "stim"]
+        delta = 0.5  # TODO: Probably should also be 0.25?
+        offset = 1
```
Comment (drammock) [inline mne/viz/tests/test_raw.py:1244]: just curious why you skip checking "grad" for the pyqtgraph browser

**Review by drammock — APPROVED** [review]
LGTM, just one question

File: mne/viz/tests/test_raw.py (hunk 1 above)
Comment (larsoner) [reply mne/viz/tests/test_raw.py:1244]: I don't think I do... `"grad"` is in the list and we `.pop(-1)` i.e. the `"stim"` channel. The others (grad, mag, eeg, eog) all get checked

Comment (drammock) [reply mne/viz/tests/test_raw.py:1244]: Oops you're right. I was thinking the nonzero second arg to enumerate was making it skip the first element (which of course isn't right)

**Pull request merged**